NGitLab/Impl/FilesClient.cs

```diff
@@ -55,6 +70,32 @@ public bool FileExists(string filePath, string @ref)
         }
     }
 
+    public Task<bool> FileExistsAsync(string filePath, string @ref, CancellationToken cancellationToken = default)
+    {
+        return _api.Head().ExecuteAsync(_repoPath + $"/files/{EncodeFilePath(filePath)}?ref={@ref}", cancellationToken)
+            .ContinueWith(t =>
```

Comment:
Please use `async`/`await`, as it make the code simpler and clearer.

Reply:
Done. Thanks for taking a look!